fast64_internal/f3d/f3d_gbi.py

```diff
@@ -2493,23 +2548,49 @@ def to_c(self, textureExportSettings: TextureExportSettings, gfxFormatter: GfxFo
         self.freePalettes()
         return ExportCData(staticData, dynamicData, texC)
 
-    def to_c_vertex_scroll(self, scrollName, gfxFormatter):
-        scrollData = CData()
-        for name, mesh in self.meshes.items():
-            scrollData.append(mesh.to_c_vertex_scroll(gfxFormatter))
+    def to_c_scroll(self, funcName: str, gfxFormatter: GfxFormatter) -> CScrollData:
+        data = CScrollData()
+        vertexScrollData = self.to_c_vertex_scroll(gfxFormatter)
+        if len(vertexScrollData.functionCalls) > 0:
+            data.append(vertexScrollData)
 
-        hasScrolling = len(scrollData.header) > 0
+        gfxScrollData = self.to_c_gfx_scroll(gfxFormatter)
+        if len(gfxScrollData.functionCalls) > 0:
+            data.append(gfxScrollData)
 
-        scrollDefinesSplit = scrollData.header.split("extern void ")
-        scrollData.source += "void scroll_" + scrollName + "() {\n"
-        for scrollFunc in scrollDefinesSplit:
-            if scrollFunc == "":
-                continue
-            scrollData.source += "\t" + scrollFunc
-        scrollData.source += "}\n"
+        data.topLevelScrollFunc = f"scroll_{funcName}"
+        data.source += f"void {data.topLevelScrollFunc}() {{\n"
+        for scrollFunc in data.functionCalls:
+            data.source += f"\t{scrollFunc}();\n"
+        data.source += f"}};\n"
```

Comment:
this doesnt really need to be an f string since there aint no formmattin i reckon

Reply:
I think it makes the curly brace balance more clear, especially with vscode syntax coloring

Reply:
yeah good point